File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/rxCache/OcmCache.java
package com.gigigo.orchextra.core.data.rxCache;

import android.content.Context;
import com.gigigo.orchextra.core.data.api.dto.content.ApiSectionContentData;
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementData;
import com.gigigo.orchextra.core.data.api.dto.menus.ApiMenuContentData;
import com.gigigo.orchextra.core.data.api.dto.versioning.ApiVersionKache;
import com.gigigo.orchextra.core.data.api.dto.video.ApiVideoData;
import io.reactivex.Observable;

public interface OcmCache {

  Observable<ApiMenuContentData> getMenus();

  void putMenus(ApiMenuContentData apiMenuContentData);

  boolean isMenuCached();

  boolean isMenuExpired();

  Observable<ApiSectionContentData> getSection(String elementUrl);

  void putSection(ApiSectionContentData apiSectionContentData);

  boolean isSectionCached(String elementUrl);

  boolean isSectionExpired(String elementUrl);

  Observable<ApiElementData> getDetail(String slug);

  void putDetail(ApiElementData apiElementData);

  boolean isDetailCached(String slug);

  boolean isDetailExpired(String slug);

  Observable<ApiVideoData> getVideo(String videoId);

  void putVideo(ApiVideoData videoData);

  boolean isVideoCached(String videoId);

  boolean isVideoExpired(String videoId);

  void evictAll(boolean images, boolean data);

  Context getContext();

  void putVersion(ApiVersionKache apiVersionKache);

  Observable<ApiVersionKache> getVersion();

  boolean isVersionCached();

  boolean isVersionExpired();
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/rxCache/OcmCache.kt
package com.gigigo.orchextra.core.data.rxCache

import android.content.Context
import com.gigigo.orchextra.core.data.api.dto.content.ApiSectionContentData
import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementData
import com.gigigo.orchextra.core.data.api.dto.menus.ApiMenuContentData
import com.gigigo.orchextra.core.data.database.entities.DbElementData
import com.gigigo.orchextra.core.data.database.entities.DbMenuContentData
import com.gigigo.orchextra.core.data.database.entities.DbSectionContentData
import com.gigigo.orchextra.core.data.database.entities.DbVideoData
import gigigo.com.vimeolibs.VimeoInfo
import io.reactivex.Observable

interface OcmCache {

  val context: Context

  fun getMenus(): Observable<DbMenuContentData>
  fun hasMenusCached(): Boolean
  fun putMenus(apiMenuContentData: ApiMenuContentData)

  fun getSection(elementUrl: String): Observable<DbSectionContentData>
  fun putSection(apiSectionContentData: ApiSectionContentData, key: String)
  fun isSectionCached(elementUrl: String): Boolean
  fun isSectionExpired(elementUrl: String): Boolean

  fun getDetail(slug: String): Observable<DbElementData>
  fun isDetailCached(slug: String): Boolean
  fun isDetailExpired(slug: String): Boolean
  fun putDetail(apiElementData: ApiElementData, key: String)

  fun getVideo(videoId: String): Observable<DbVideoData>
  fun isVideoCached(videoId: String): Boolean
  fun isVideoExpired(videoId: String): Boolean
  fun putVideo(videoData: VimeoInfo)

  fun evictAll(images: Boolean, data: Boolean)
}